ffi/src/value/results.rs

```diff
@@ -195,6 +197,76 @@ pub enum ProposalResult<'db> {
     Err(OwnedBytes),
 }
 
+/// A result type returned from FFI functions that create an iterator
+#[derive(Debug)]
+#[repr(C)]
+pub enum IteratorResult<'db> {
+    /// The caller provided a null pointer to a database handle.
+    NullHandlePointer,
+    /// Building the proposal was successful and the proposal ID and root hash
+    /// are returned.
+    Ok {
+        /// An opaque pointer to the [`ProposalHandle`] that can be use to create
+        /// an additional proposal or later commit. The caller must ensure that this
+        /// pointer is freed with [`fwd_free_proposal`] if it is not committed.
+        ///
+        /// [`fwd_free_proposal`]: crate::fwd_free_proposal
+        // note: opaque pointers mut be boxed because the FFI does not the structure definition.
+        handle: Box<IteratorHandle<'db>>,
+    },
+    /// An error occurred and the message is returned as an [`OwnedBytes`]. If
+    /// value is guaranteed to contain only valid UTF-8.
+    ///
+    /// The caller must call [`fwd_free_owned_bytes`] to free the memory
+    /// associated with this error.
+    ///
+    /// [`fwd_free_owned_bytes`]: crate::fwd_free_owned_bytes
+    Err(OwnedBytes),
+}
+
+/// A result type returned from iterator FFI functions
+#[derive(Debug)]
+#[repr(C)]
+pub enum KeyValueResult {
+    /// The caller provided a null pointer to an iterator handle.
+    NullHandlePointer,
+    /// The iterator returned empty result, the iterator is exhausted
+    None,
+    /// The next item on iterator is returned.
+    Some(OwnedKeyValuePair),
+    /// An error occurred and the message is returned as an [`OwnedBytes`]. If
+    /// value is guaranteed to contain only valid UTF-8.
```

Comment:
I don't understand this. If the word "if" at the beginning is extra, then I disagree with this assertion. If this was about to say what happens in that case, then please complete the sentence.